stdlib/list.ml

```diff
@@ -244,6 +244,13 @@ let find_all p =
 
 let filter = find_all
 
+let filteri p =
```

Comment:
The current implementations in List are incoherent in this respect, but I find it more readable to eta-expand, i.e. declare the `l` argument and pass it to `aux` explicitly.  A micro-benchmark suggests this is also a bit more efficient (around 9% with a trivial predicate and an empty list).